sormas-backend/src/main/java/de/symeda/sormas/backend/person/PersonFacadeEjb.java

```diff
@@ -1859,6 +1866,84 @@ public void mergePerson(String leadPersonUuid, String otherPersonUuid, boolean m
 		service.ensurePersisted(leadPerson);
 	}
 
+	private void processPersonAddressMerge(PersonDto leadPersonDto, PersonDto otherPersonDto) {
+		if (differentLocation(leadPersonDto.getAddress(), otherPersonDto.getAddress())) {
+			LocationDto otherAddress = otherPersonDto.getAddress();
+			otherAddress.setAddressType(PersonAddressType.OTHER_ADDRESS);
+			otherAddress.setAddressTypeDetails(I18nProperties.getString(Strings.messagePersonMergedAddressDescription));
+			leadPersonDto.addAddress(otherAddress);
+		} else {
+			DtoHelper.copyDtoValues(leadPersonDto.getAddress(), otherPersonDto.getAddress(), false);
+		}
+	}
+
+	private boolean differentLocation(LocationDto firstAddress, LocationDto secondAddress) {
+
+		java.util.function.Predicate<Function<LocationDto, HasUuid>> differentInfrastructurePredicate = (Function<LocationDto, HasUuid> function) -> {
+			return function.apply(firstAddress) != null
+				&& function.apply(secondAddress) != null
+				&& !DataHelper.equal(function.apply(firstAddress).getUuid(), function.apply(secondAddress).getUuid());
+
+		};
+
+		if (Boolean.TRUE.equals(differentInfrastructurePredicate.test(LocationDto::getCountry))) {
+			return true;
+		}
+
+		if (Boolean.TRUE.equals(differentInfrastructurePredicate.test(LocationDto::getRegion))) {
+			return true;
+		}
+
+		if (Boolean.TRUE.equals(differentInfrastructurePredicate.test(LocationDto::getDistrict))) {
+			return true;
+		}
+
+		if (Boolean.TRUE.equals(differentInfrastructurePredicate.test(LocationDto::getCommunity))) {
+			return true;
+		}
+
+		if (firstAddress.getFacility() != null) {
+			FacilityDto firstAddressFacilityDto = facilityFacade.getByUuid(firstAddress.getFacility().getUuid());
+			if (secondAddress.getCommunity() != null
+				&& !firstAddressFacilityDto.getCommunity().getUuid().equals(secondAddress.getCommunity().getUuid())) {
```

Comment:
Communities are optional for facilities, so this can still result in a NPE (because you're not checking whether the facility of the first address has one)